uaclient/cli.py

```diff
@@ -1508,7 +1509,9 @@ def get_parser():
     return parser
 
 
+@set_event_mode
 def action_status(args, *, cfg):
+    event.set_command("status")
```

Comment:
The decorator which sets it to machine-readable mode is applied only when the action function is called, so if any error happens before that it will be printed out as CLI (non structured, plain text) output - this applies to the other commands as well

Reply:
Yes, but if we have an exception, like when setting up the proxy for example, shouldn't we return this as a json response in that situation ?

Reply:
I understand how it can fail there, and although it is pretty much an unexpected case, yes, I agree that we should put out JSON if JSON was requested no matter what.
Then, we need to set the event logger mode on main as well, instead of relying on the decorator to do it. We will already know the command and the requested format.
I will move the logic there to make sure it goes well.

Reply:
@lucasmoura The earliest that we can do it is just after parsing the args.  The change is in a separate commit to facilitate reviewing/changing/removing as needed.